client/me/account/main.jsx

```diff
@@ -98,29 +108,28 @@ const INTERFACE_FIELDS = [
 const Account = createReactClass( {
 	displayName: 'Account',
 
-	mixins: [ observe( 'userSettings' ) ],
-
-	propTypes: {
-		userSettings: PropTypes.object.isRequired,
-	},
-
 	UNSAFE_componentWillMount() {
 		// Clear any username changes that were previously made
 		this.clearUsernameValidation();
-		this.props.userSettings.removeUnsavedSetting( 'user_login' );
+		this.props.removeUnsavedUserSetting( 'user_login' );
 	},
 
 	componentDidMount() {
 		debug( this.constructor.displayName + ' component is mounted.' );
-		this.props.userSettings.getSettings();
 		this.debouncedUsernameValidate = debounce( this.validateUsername, 600 );
 	},
 
+	componentDidUpdate() {
+		if ( ! this.hasUnsavedUserSettings( [].concat( ACCOUNT_FIELDS, INTERFACE_FIELDS ) ) ) {
```

Comment:
Nit: is there any difference between `[].concat( A, B )` and `A.concat( B )`? Do we earn anything by starting with the empty array first?

Reply:
The latter is probably better because it avoids the creation of an empty array. Changed it here https://github.com/Automattic/wp-calypso/pull/49760/commits/66019a4b5e227f79fb9b0a826b1336c5a655c5b7.